lib/queuePopulator/KafkaLogConsumer/ListRecordStream.js

```diff
@@ -50,6 +50,29 @@ class ListRecordStream extends stream.Transform {
         }
     }
 
+    /**
+     * Get object metadata from oplog event
+     * @param {ChangeStreamDocument} changeStreamDocument changeStreamDocument
+     * @returns {string|undefined} stringified objectMd
+     */
+    _getObjectMd(changeStreamDocument) {
+        // Metadata values from updateDescription are used instead of
+        // fullDocument as they represent the exact changes that occured
+        // to the object. We only need this for the update operation.
+
+        // The fullDocument value contains a copy of the entire updated
+        // document at a point in time after the update, this can cause issues
+        // when the object gets deleted just after the update (only for updates).
+        const updateDescriptionMd = changeStreamDocument.updateDescription
+            && changeStreamDocument.updateDescription.updatedFields &&
+            changeStreamDocument.updateDescription.updatedFields.value;
+        if (changeStreamDocument.operationType === 'update') {
+            return updateDescriptionMd && JSON.stringify(updateDescriptionMd);
+        }
+        const fullDocumentMd = changeStreamDocument.fullDocument && changeStreamDocument.fullDocument.value;
```

Comment:
not relevant anymore since `change.stream.full.document` is disabled

Reply:
yes `insert` operations which we do when first creating an object only have the fullDocument field